xml_struct/src/tests.rs

```diff
@@ -173,3 +176,67 @@ fn int_as_element() {
         "Serializing `u64` should result in bare text content"
     );
 }
+
+#[test]
+fn bool() {
+    // Test `true` as a content node.
+    let content = true;
+
+    let expected = "true";
+    let actual = serialize_value_children(content).expect("Failed to serialize value");
+
+    assert_eq!(
+        actual, expected,
+        "`true` should be serialized as string value 'true'"
+    );
+
+    // Test `false` as a content node.
+    let content = false;
+
+    let expected = "false";
+    let actual = serialize_value_children(content).expect("Failed to serialize value");
+
+    assert_eq!(
+        actual, expected,
+        "`false` should be serialized as string value 'false'"
+    );
+
+    let element_name = "foo";
+    let attr_name = "bar";
+
+    // Test `true` as an attribute value.
+    let content = true;
+
+    let expected = vec![(attr_name, "true").into()];
```

Comment:
suggestion: it's not obvious from reading the code what the target type of `into()` is.

```suggestion
    let expected: Vec<Attribute> = vec![(attr_name, "true").into()];
```